custom/enikshay/nikshay_datamigration/management/commands/create_enikshay_cases.py

```diff
@@ -0,0 +1,170 @@
+from django.core.management import BaseCommand
+from django.db import transaction
+
+from casexml.apps.case.const import CASE_INDEX_EXTENSION
+from casexml.apps.case.mock import CaseFactory, CaseStructure, CaseIndex
+from corehq.apps.locations.models import SQLLocation
+
+from custom.enikshay.nikshay_datamigration.models import PatientDetail, Outcome, Followup
+from dimagi.utils.decorators.memoized import memoized
+
+ENIKSHAY_DOMAIN = 'enikshay-np'
+
+
+class EnikshayCaseFactory(object):
+
+    patient_detail = None
+
+    def __init__(self, patient_detail, nikshay_code_to_location):
+        self.patient_detail = patient_detail
+        self.factory = CaseFactory(domain=ENIKSHAY_DOMAIN)
+        self.nikshay_code_to_location = nikshay_code_to_location
```

Comment:
I wanted to cache this since finding the location with matching nikshay ID is slow (the 3rd party library doesn't support lookups on json fields).

Could structure this in another way though, like by moving to a classmethod or using quickcache.